src/Development/IDE/LSP/Outline.hs

```diff
@@ -167,6 +167,28 @@ documentSymbolForDecl (L l (ForD x)) = Just
 
 documentSymbolForDecl _ = Nothing
 
+-- | Wrap the Document imports into a hierarchical outline for 
+-- a better overview of symbols in scope. 
+-- If there are no imports, then no hierarchy will be created.
+documentSymbolForImportSummary :: [DocumentSymbol] -> [DocumentSymbol]
+documentSymbolForImportSummary [] = []
+documentSymbolForImportSummary importSymbols = 
+    let 
+      importRange = foldr (\DocumentSymbol {..} (Range accStart accEnd) -> 
+        let Range start end = _range 
+        in  Range (min start accStart) (max end accEnd)
+        ) 
+        (Range (Position maxBound 0) (Position 0 0))
+        importSymbols
```

Comment:
I dislike this code. Is there something nicer for it?

Reply:
```
-- safe because if we have no ranges then we don't take this branch
mergeRanges xs = Range (minimum $ map _start xs) (maximum $ map _end xs)
importRange = mergeRanges $ map _range importSymbols
```